src/nvim/option.c

```diff
@@ -2436,6 +2436,11 @@ static bool valid_filetype(char_u *val)
   return true;
 }
 
+#ifdef _MSC_VER
+// MSVC optimizations are disabled for this function because it
+// incorrectly generates an empty string for SHM_ALL.
```

Comment:
The optimizer probably uses an empty string because the old pointer became invalid.

Problem seems to be that the use of `SHM_ALL` creates a local array that is assigned to pointer `p` (line 3201), but after leaving the block scope, `p` is used to access the (destroyed) array (line 3213).